app/src/main/java/com/example/cocktails/presentation/fragments/details/CocktailDetailsFragment.kt

```diff
@@ -0,0 +1,162 @@
+                    val cocktailDetails = resource.data
+                    cocktailDetailsMap = getNonNullFields(cocktailDetails)
+                    createList(cocktailDetails)
+                    setDetails()
+                    initUI()
+                }
+
+                is Resource.Loading -> {
+                    binding.progressBar.visibility = View.VISIBLE
+                }
+
+                is Resource.Error -> {
+                    binding.progressBar.visibility = View.GONE
+
+                    val builder: AlertDialog.Builder = AlertDialog.Builder(context)
+                    builder.setMessage(resources.getString(R.string.filter_error, resource.message))
+                        .setTitle("Error")
+                        .setPositiveButton("OK") { _, _ ->
+                        }
+
+                    val dialog: AlertDialog = builder.create()
+                    dialog.show()
+                }
+            }
+        }
+
+        cocktailDetailsViewModel.fetchAllCocktailDetails(argId)
+
+    }
+
+    private fun createList(cocktailDetails: List<CocktailDetails>) {
+        cocktailList = cocktailDetails.map { cocktailDetailsItem ->
+            Cocktail(
+                id = cocktailDetailsItem.idDrink,
+                name = cocktailDetailsItem.strDrink,
+                image = cocktailDetailsItem.strDrinkThumb,
+                alcoholic = cocktailDetailsItem.strAlcoholic,
+                favorite = argFavorite,
+                email = argEmail
+            )
+        }
+    }
+
+    private fun initUI() {
+        Glide.with(binding.cocktailImage.context)
+            .load(cocktailDetailsMap?.get("strDrinkThumb"))
+            .into(binding.cocktailImage)
+
+        binding.cocktailName.text = cocktailDetailsMap?.get("strDrink")
+        binding.labelAlcoholic.text = cocktailDetailsMap?.get("strAlcoholic")
+        binding.labelCategory.text = cocktailDetailsMap?.get("strCategory")
+        binding.labelGlass.text = cocktailDetailsMap?.get("strGlass")
+        binding.ingredientsItems.text = stringBuilderIngredient
+        binding.instructionsItems.text = cocktailDetailsMap?.get("strInstructions")
+        setImage()
+
+        binding.favoriteButton.setOnClickListener {
+            if (argFavorite) {
+                cocktailDetailsViewModel.deleteCocktail(cocktailList.first())
+            } else {
+                cocktailDetailsViewModel.insertCocktail(cocktailList.first())
+            }
+            argFavorite = !argFavorite
+            setImage()
+        }
+    }
+
+    private fun setImage() {
+        val imageResourceId = if (argFavorite) {
+            R.drawable.imageview_checked
+        } else {
+            R.drawable.imageview_unchecked
+        }
+        Glide.with(binding.favoriteButton.context).load(imageResourceId)
+            .into(binding.favoriteButton)
+    }
+
+    private fun getNonNullFields(cocktailDetailsList: List<CocktailDetails>): Map<String, String>? {
```

Comment:
Can we move this and the next method setDetails() inside a view model? Since Reflection can be long running task we should try to do it on the background thread. This means that we can do all of this after successful fetch. We can also move the needed map to a viewModel and make it public so we can access it from fragment after it is filled with data.